Eatery Blue/UI/HomeScreen/HomeModelController.swift

```diff
@@ -206,26 +206,25 @@ class HomeModelController: HomeViewController {
                     cells.append(.carouselView(carouselView))
                 }
 
-                if !allEateries.isEmpty {
+                currentEateries = allEateries
+                if !currentEateries.isEmpty {
                     cells.append(.titleLabel(title: "All Eateries"))
                 }
 
-                eateryStartIndex = cells.count // track the index of the first eateryCard in cells
-                for eatery in allEateries {
-                    cells.append(.eateryCard(eatery: eatery))
-                }
-
             } else {
                 let predicate = filter.predicate(userLocation: LocationManager.shared.userLocation, departureDate: Date())
                 let filteredEateries = allEateries.filter({
                     predicate.isSatisfied(by: $0, metadata: coreDataStack.metadata(eateryId: $0.id))
                 })
-                for eatery in filteredEateries {
-                    cells.append(.eateryCard(eatery: eatery))
-                }
+                currentEateries = filteredEateries
+            }
+
+            eateryStartIndex = cells.count // track the index of the first eateryCard in cells
+            for eatery in currentEateries {
+                cells.append(.eateryCard(eatery: eatery))
             }
```

Comment:
Can you use the `forEach` HOF here for simplicity?
`currentEateries.forEach { eatery in
    cells.append(.eateryCard(eatery: eatery))
}`